### PR TITLE
Fix: #2471. Clarify status code text.

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- clarifies text around supported status codes.

Fixes: #2471

### CODE REVIEW

**Comment by ioggstream** [comment]
@MikeRalphson as you requested :)

**Review by MikeRalphson — DISMISSED** [review]
Thanks @ioggstream! Please could you retarget against `versions/3.1.1.md` on the `v3.1.1-dev` branch?

File: versions/3.1.1.md [hunk 1]
```diff
@@ -97,7 +97,7 @@ Some examples of possible media type definitions:
 ```
 ##### <a name="httpCodes"></a>HTTP Status Codes
 The HTTP Status Codes are used to indicate the status of the executed operation. 
-The available status codes are defined by [RFC7231](https://tools.ietf.org/html/rfc7231#section-6) and registered status codes are listed in the [IANA Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml).
+Status codes are defined in [RFC7231](https://tools.ietf.org/html/rfc7231#section-6) and the available status codes are registered in the [IANA Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml).
```
Comment (darrelmiller) [inline versions/3.1.1.md:100]:
```suggestion
Status codes SHOULD be selected from the available status codes registered in the [IANA Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml).
```

**Comment by karenetheridge** [comment]
This suggests the JSON Schema should be softened to support more than just `[1-5][0-9][0-9]`. Can you make that change as well? (schemas/v3.1/schema.json and .yaml)

**Comment by ioggstream** [comment]
Hi @karenetheridge,  maybe I'm missing something. 
The IANA table covers the range `[100; 599]` as specified in https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#status.codes

> All valid status codes are within the range of 100 to 599, inclusive

iiuc all IANA status codes should comply with the above spec, and that there should be no change in  https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/schema.yaml#L529


If its not  the case, how would you modify the above file?

thx++, R

**Comment by darrelmiller** [comment]
@karenetheridge There are no status codes in the IANA registry outside of the 100-599 range, so the regex in JSON Schema is fine.

**Review by darrelmiller — APPROVED** [review]

**Comment by karenetheridge** [comment]
The specification says SHOULD -- so doesn't this mean that it is still permissible for values to be outside that range (for some unusual usecase)?

**Pull request merged**